src/test/java/org/folio/rest/api/BoundWithStorageTest.java

```diff
@@ -0,0 +1,139 @@
+package org.folio.rest.api;
+
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import junitparams.JUnitParamsRunner;
+import org.folio.rest.support.IndividualResource;
+import org.folio.rest.support.Response;
+import org.folio.rest.support.builders.HoldingRequestBuilder;
+import org.folio.rest.support.builders.ItemRequestBuilder;
+import org.folio.rest.support.http.ResourceClient;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.net.HttpURLConnection;
+import java.util.List;
+import java.util.UUID;
+
+import static org.folio.rest.support.http.InterfaceUrls.*;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+@RunWith(JUnitParamsRunner.class)
+public class BoundWithStorageTest extends TestBaseWithInventoryUtil {
+  static ResourceClient boundWithPartsClient  = ResourceClient.forBoundWithParts(client);
+
+  @After
+  public void afterEach() {
+    deleteBoundWithParts();
+    StorageTestSuite.deleteAll(itemsStorageUrl(""));
+    StorageTestSuite.deleteAll(holdingsStorageUrl(""));
+    StorageTestSuite.deleteAll(instancesStorageUrl(""));
+  }
+
+  @Test
+  public void canCreateAndRetrieveBoundWithParts() {
+    IndividualResource mainInstance = createInstance("Main Instance");
+    IndividualResource mainHoldingsRecord = createHoldingsRecord(mainInstance.getId());
+    IndividualResource item = createItem(mainHoldingsRecord.getId());
+
+    IndividualResource anotherInstance = createInstance("Another Instance");
+    IndividualResource anotherHoldingsRecord = createHoldingsRecord(anotherInstance.getId());
+    IndividualResource aThirdInstance = createInstance("Third Instance");
+    IndividualResource aThirdHoldingsRecord = createHoldingsRecord(aThirdInstance.getId());
+
+    // Make 'item' a bound-with
+    boundWithPartsClient.create(createBoundWithPartJson(mainHoldingsRecord.getId(),item.getId()));
+    IndividualResource secondPart = boundWithPartsClient.create(createBoundWithPartJson(anotherHoldingsRecord.getId(),item.getId()));
+    boundWithPartsClient.create(createBoundWithPartJson(aThirdHoldingsRecord.getId(),item.getId()));
+
+    Response boundWithGETResponseForPartById = boundWithPartsClient.getById(secondPart.getId());
+
+    List<JsonObject> getAllPartsForBoundWithItem = boundWithPartsClient.getByQuery("?query=itemId==" + item.getId());
+
+    assertThat(boundWithGETResponseForPartById.getStatusCode(), is(HttpURLConnection.HTTP_OK));
+    assertThat(getAllPartsForBoundWithItem.size(), is(3));
```

Comment:
This assertions suggest we only care that there are 3 bound-with entries for this item and not anything else about those entries, is that the case?

Reply:
Yes. there are not any properties in these, it's all primary key and foreign keys.